packages/lib/src/nav-tabs/NavTabs.tsx

```diff
@@ -1,79 +1,32 @@
-import { Children, KeyboardEvent, ReactElement, ReactNode, useEffect, useMemo, useRef, useState } from "react";
+import { Children, KeyboardEvent, ReactElement, useMemo, useState } from "react";
 import styled from "styled-components";
 import NavTabsPropsType from "./types";
-import DxcTab from "./Tab";
+import Tab from "./Tab";
 import NavTabsContext from "./NavTabsContext";
+import { getLabelFromTab, getPropInChild, getPreviousTabIndex, getNextTabIndex } from "./utils";
 
-const getPropInChild = (child: ReactNode, propName: string) => {
-  if (child && typeof child === "object" && "props" in child) {
-    const childWithProps = child as ReactElement;
-    if (childWithProps.props[propName]) {
-      return childWithProps.props[propName];
-    } else if (childWithProps.props.children) {
-      return getPropInChild(childWithProps.props.children, propName);
-    }
-  }
-};
-
-const getLabelFromTab = (child: ReactNode) => {
-  if (typeof child === "string") {
-    return child;
-  } else if (child && typeof child === "object" && "props" in child) {
-    const childWithProps = child as ReactElement;
-    if (Array.isArray(childWithProps.props.children)) {
-      return getLabelFromTab(childWithProps.props.children[0]);
-    } else {
-      return getLabelFromTab(childWithProps.props.children);
-    }
-  }
-};
-
-const getPreviousTabIndex = (array: ReactElement[], initialIndex: number): number => {
-  let index = initialIndex === 0 ? array.length - 1 : initialIndex - 1;
-  while (getPropInChild(array[index], "disabled")) {
-    index = index === 0 ? array.length - 1 : index - 1;
-  }
-  return index;
-};
-
-const getNextTabIndex = (array: ReactElement[], initialIndex: number): number => {
-  let index = initialIndex === array.length - 1 ? 0 : initialIndex + 1;
-  while (getPropInChild(array[index], "disabled")) {
-    index = index === array.length - 1 ? 0 : index + 1;
-  }
-  return index;
-};
+const NavTabsContainer = styled.div`
+  position: relative;
+  display: flex;
+  overflow: auto hidden;
+`;
 
-const Underline = styled.div<{ underlineWidth: number }>`
+const Underline = styled.div`
   position: absolute;
   bottom: 0;
   left: 0;
   height: var(--border-width-m);
   background-color: var(--border-color-neutral-medium);
-  z-index: -1;
-  width: ${(props) => props.underlineWidth}px;
+  width: 100%;
```

Comment:
this change causes issues when the nav tabs don't have enough space
![image](https://github.com/user-attachments/assets/69ef9983-145a-4f36-93b1-6346b0d50d9d)

Reply:
Solved!